pyproject.toml

```diff
@@ -59,11 +59,16 @@ pinned = [ # Pinned versions of core dependencies
 ]
 
 typing = [
+  'IPython<8.31.0',
+  'imageio<2.37.0',
+  'ipywidgets<8.2.0',
   'mypy<1.14.0',
   'npt-promote==0.2',
   'numpy>=2.0.0',
+  'pandas-stubs<2.2.4',
   'pyvista[pinned]',
   'trimesh<4.6.0',
+  'types-tqdm<4.67.0',
```

Comment:
`mypy` initially complained about many packages not being imported (`import-not-found`).
But, installation does not fix the issue in many cases, because if the package is untyped,  then an error is still generated:
``` python
Skipping analyzing "<module name>": module is installed, but missing library stubs or py.typed marker  [import-untyped]
```

I have found that the following packages are _not_ marked as typed, and therefore there is no value in having them as a typing dependency, and any of these imports _must_ be ignored:
`sympy`, `meshio`, `scipy`, `pooch` `colorcet`, `cmocean`, `sphinx_gallery`, `trame`, `nest_asyncio`,`osmnx`,

But, after installing _these_ packages, `mypy` no longer complains about missing imports: 
  'IPython<8.31.0',
  'imageio<2.37.0',
  'ipywidgets<8.2.0',
  'pandas-stubs<2.2.4',
  'types-tqdm<4.67.0',

And so If the goal is to minimize the number of `type: ignore` comments, then we should add these as new dependencies.